[DON'T MERGE] [feat]VSWA support for PyTorch KVCacheManager

qixiang-99 opened the pull request:
# [DON'T MERGE] [feat] Add VSWA support for PyTorch KVCacheManager

This PR depends on [PR4667](https://github.com/NVIDIA/TensorRT-LLM/pull/4667), and the code may change since PR4667 is not yet merged.

Pending tasks:

- [ ] Clean up logging code
- [ ] Add detailed description
- [ ] Update code after merging [PR4667](https://github.com/NVIDIA/TensorRT-LLM/pull/4667)

## Description

Please explain the issue and the solution in short.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

qixiang-99 committed:
Refactor and pipeclean, can pass gemma3 test. Need to run CI to find out whether there are more issues.

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

netanel-haber committed:
Fix underallocation in VSWA+/VGQA

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

tensorrt-cicd commented:
[PR_Github #7961](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/7961/) [ run ] triggered by Bot

qixiang-99 committed:
fix:
- Many tests directly pass executor.KvCacheConfig to KvCacheManager, apply a conversion if this happen
- add `self.blocks_in_primary_pool` and `self.blocks_in_secondary_pool` as these are used for standard case.

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

qixiang-99 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #7994](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/7994/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #5793](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/5793/) completed with status: 'FAILURE'

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

netanel-haber committed:
fix minor bug

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

netanel-haber committed:
actually use reduced value [blocksWorld] and assign it to blocksPrimary and blocksSecondary

Signed-off-by: Netanel Haber <nhaber@nvidia.com>

qixiang-99 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #8046](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8046/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #8046](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8046/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #5832](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/5832/) completed with status: 'FAILURE'

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

qixiang-99 committed:
fix: correct the test that passes the kvcache config with the wrong type (the mirror class of executor.kvcache_config)

Signed-off-by: qixiang-99 <203170375+qixiang-99@users.noreply.github.com>

qixiang-99 commented:
/bot run --disable-fail-fast

netanel-haber committed:
Merge branch 'main' into user/nhaber/fix-variable-window-size-underallocation

Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #8149](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8149/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #5906](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/5906/) completed with status: 'FAILURE'

qixiang-99 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #8240](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8240/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #8240](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/8240/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #5972](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/5972/) completed with status: 'FAILURE'
